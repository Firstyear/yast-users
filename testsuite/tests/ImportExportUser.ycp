/**
 * File:
 *  ImportExportUser.ycp
 *
 * Module:
 *  Users configurator
 *
 * Summary:
 *  converting user maps: autoyast -> users module -> autoyast
 *
 * Authors:
 *  Jiri Suchomel <jsuchome@suse.cz>
 *
 */

{

    include "testsuite.ycp";
    import "Users";

    map importing = $[
	"encrypted": true,
	"forename":"aa",
	"surname":"A",
	"uid":500,
        "gid":500,
	"home":"/home/aaa",
	"user_password":"5OkhN2nLxMfEY",
	"password_settings":$[
        "inact":-1,
        "max":99999,
        "min":0,
        "warn":7
        ],
        "grouplist":"audio",
	"shell":"/bin/bash",
	"username":"aaa"
    ];

    DUMP ( sformat ("importing:\n %1", importing) );

    DUMP ("==== processing ImportUser... ============================");

    map user = TEST(``(Users::ImportUser (importing)), [], 0);

    DUMP ("==== processing ExportUser... ============================");

    TEST(``(Users::ExportUser (user)), [], 0);

}
