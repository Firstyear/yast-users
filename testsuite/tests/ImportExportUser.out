Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp/"
Read	.target.tmpdir "/tmp/"
Execute	.target.bash "/usr/lib/YaST2/bin/version" 0
Read	.sysconfig.language.RC_LANG ""
Read	.sysconfig.language.DEFAULT_LANGUAGE ""
Read	.target.yast2 "lang2iso.ycp" nil
Read	.target.size "/etc/sysconfig/language" -1
Read	.sysconfig.console.CONSOLE_FONT ""
Read	.sysconfig.console.CONSOLE_SCREENMAP ""
Read	.sysconfig.console.CONSOLE_UNICODEMAP ""
Read	.sysconfig.console.CONSOLE_MAGIC ""
Read	.sysconfig.console.CONSOLE_ENCODING ""
Read	.sysconfig.language.RC_LANG ""
Dump	importing:
Dump	 $["encrypted":true, "forename":"aa", "gid":500, "grouplist":"audio", "home":"/home/aaa", "password_settings":$["inact":-1, "max":99999, "min":0, "warn":7], "shell":"/bin/bash", "surname":"A", "uid":500, "user_password":"5OkhN2nLxMfEY", "username":"aaa"]
Dump	==== processing ImportUser... ============================
Execute	.target.bash_output "date +%s" $[]
Read	.target.size "" nil
Return	$["encrypted":true, "fullname":"aa A", "gid":500, "grouplist":"audio", "home":"/home/aaa", "modified":"imported", "password":"5OkhN2nLxMfEY", "shadow":$["inact":-1, "max":99999, "min":0, "password":"5OkhN2nLxMfEY", "warn":7], "shell":"/bin/bash", "type":"local", "uid":500, "username":"aaa"]
Dump	==== processing ExportUser... ============================
Return	$["encrypted":true, "forename":"aa", "fullname":"aa A", "gid":500, "grouplist":"audio", "home":"/home/aaa", "password_settings":$["inact":-1, "max":99999, "min":0, "warn":7], "shell":"/bin/bash", "surname":"A", "uid":500, "user_password":"5OkhN2nLxMfEY", "username":"aaa"]
